### PR TITLE
[6/N] Don't skip ASAN on some tests 

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: aten/src/ATen/native/Scalar.cpp [hunk 1]
```diff
@@ -35,6 +35,9 @@ Scalar item(const Tensor& self) {
 #endif
 
 Scalar _local_scalar_dense_cpu(const Tensor& self) {
+  if (self.scalar_type() == kBool) {
+    return Scalar(static_cast<bool>(*reinterpret_cast<const uint8_t*>(self.const_data_ptr<bool>())));
```
Comment (ezyang) [inline aten/src/ATen/native/Scalar.cpp:39]: What's going on here?

Comment (cyyever) [reply aten/src/ATen/native/Scalar.cpp:39]: ASAN fails when load  a bool value out of range. Therefore, we case a bool pointer to uint8, de-reference it, and then cast to bool.

Comment (cyyever) [reply aten/src/ATen/native/Scalar.cpp:39]: Because of this ASAN error, many tests are marked to skip ASAN.

Comment (ezyang) [reply aten/src/ATen/native/Scalar.cpp:39]: A comment on this would be good

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**